spec/features/result_filters/location_spec.rb

```diff
@@ -0,0 +1,167 @@
+    it "Preselects across england" do
+      filter_page.load(query: { l: 2 })
+      expect(filter_page.across_england.checked?).to eq(true)
+    end
+
+    it "Removes the lat filter" do
+      filter_page.load(query: { lat: "yes" })
+      filter_page.across_england.click
+      filter_page.find_courses.click
+
+      expect_page_to_be_displayed_with_query(
+        page: results_page,
+        expected_query_params: {
+          "l" => "2",
+        },
+      )
+    end
+
+    it "Removes the lng filter" do
+      filter_page.load(query: { lng: "yes" })
+      filter_page.across_england.click
+      filter_page.find_courses.click
+
+      expect_page_to_be_displayed_with_query(
+        page: results_page,
+        expected_query_params: {
+          "l" => "2",
+        },
+      )
+    end
+
+    it "Removes the rad filter" do
+      filter_page.load(query: { rad: "yes" })
+      filter_page.across_england.click
+      filter_page.find_courses.click
+
+      expect_page_to_be_displayed_with_query(
+        page: results_page,
+        expected_query_params: {
+          "l" => "2",
+        },
+      )
+    end
+
+    it "Removes the query filter" do
+      filter_page.load(query: { query: "yes" })
+      filter_page.across_england.click
+      filter_page.find_courses.click
+
+      expect_page_to_be_displayed_with_query(
+        page: results_page,
+        expected_query_params: {
+          "l" => "2",
+        },
+      )
+    end
+
+    it "Removes the loc filter" do
+      filter_page.load(query: { loc: "yes" })
+      filter_page.across_england.click
+      filter_page.find_courses.click
+
+      expect_page_to_be_displayed_with_query(
+        page: results_page,
+        expected_query_params: {
+          "l" => "2",
+        },
+      )
+    end
+
+    it "Removes the lq filter" do
+      filter_page.load(query: { lq: "yes" })
+      filter_page.across_england.click
+      filter_page.find_courses.click
+
+      expect_page_to_be_displayed_with_query(
+        page: results_page,
+        expected_query_params: {
+          "l" => "2",
+        },
+      )
+    end
+  end
+
+  describe "Validation" do
+    it "Displays an error if no option is selected" do
+      filter_page.load
+      filter_page.find_courses.click
+
+      expect(filter_page).to have_error
+    end
+  end
+
+  describe "QS parameters" do
+    it "passes querystring parameters to results" do
+      filter_page.load(query: { test: "value" })
+      filter_page.across_england.click
+      filter_page.find_courses.click
+
+      expect_page_to_be_displayed_with_query(
+        page: results_page,
+        expected_query_params: {
+          "l" => "2",
+          "test" => "value",
+        },
+      )
+    end
+
+    it "passes arrays correctly" do
+      url_with_array_params = "#{filter_page.url}?test[]=1&test[]=2"
+      PageObjects::Page::ResultFilters::Location.set_url(url_with_array_params)
+
+      filter_page.load
```

Comment:
```suggestion
      filter_page.load("test%5B%5D" => [1, 2])
```

And with the changes mentioned below in the SitePrism page, we shouldn't need the two lines above (149 and 150 currently).